### PR TITLE
#164: fix vulnerability in immer

### LINKED ISSUE #164: Prototype pollution in immer dependency
Deox currently depends on _redux-starter-kit_, which depends on a version of _immer_ for which a high-severity security issue was reported (see [Github advisory](https://github.com/advisories/GHSA-9qmh-276g-x5pj)).

The solution is to switch to the latest version of _@reduxjs/toolkit_ (the successor of _redux-starter-kit)._

### PR DESCRIPTION
**Pull request by Michael-1**
Fixes #164 
- Switch from Redux Starter Kit to its successor Redux Toolkit
- Update development dependencies that are required for the switch, or that have known security vulnerabilities
- Update configurations and tests to work with new dependencies

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Deox Changelog
 
+## [3.3.2](https://github.com/thebrodmann/deox/compare/v3.3.0...v3.3.2) (2021-03-11)
+
+-   Replace _redux-starter-kit_ with its successor @reduxjs/toolkit to fix a critical security issue in the immer dependency (see [Github advisory](https://github.com/advisories/GHSA-9qmh-276g-x5pj)).
+
```
Comment (the-dr-lazy) [inline CHANGELOG.md:6]: Changelogs generate by `sematic-release`. I think we should not change it by hand maybe!

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "deox",
-    "version": "3.3.1",
+    "version": "3.3.2",
```
Comment (the-dr-lazy) [inline package.json:3]: This should done by `semantic-release`.

File: package.json [hunk 3]
```diff
@@ -33,27 +33,26 @@
...
         "@types/node": "^10.12.18",
         "codecov": "^3.1.0",
-        "docz": "^1.0.0-rc.7",
-        "docz-theme-default": "^1.0.0-rc.7",
+        "docz": "^2.3.1",
         "dts-jest": "^23.0.0",
         "husky": "^2.0.0",
         "in-publish": "^2.0.0",
-        "jest": "^24.0.0",
+        "jest": "^26.6.3",
```
Comment (the-dr-lazy) [inline package.json:55]: Are these package updates really relevant?!

File: package.json [hunk 4]
```diff
@@ -65,14 +64,14 @@
...
+        "semantic-release": "^17.4.2",
+        "ts-jest": "^26.5.3",
         "tslint": "^5.12.1",
-        "typescript": "~3.5.2"
+        "typescript": "^4.2.3"
     },
     "peerDependencies": {
         "rxjs": ">=6.0.0-beta.0 <7",
-        "tslib": "^1.9.0"
+        "tslib": "^2.1.0"
```
Comment (the-dr-lazy) [inline package.json:74]: Also, are these relevant?

File: tsconfig.spec.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "extends": "./tsconfig.json",
     "include": ["./src/**/__tests__/*"],
     "compilerOptions": {
-        "target": "esnext",
+        "target": "es2019",
```
Comment (the-dr-lazy) [inline tsconfig.spec.json:5]: This change doesn't looks like to be relevant to the PR. I'm curious to know the reason behind this change too.

File: src/__tests__/types.dts.spec.ts [hunk 6]
```diff
@@ -28,7 +28,7 @@ displayType<Immutable<User>>()
 // @dts-jest:pass:snap
 displayType<Immutable<undefined[]>>()
 // @dts-jest:pass:snap
-displayType<Immutable<null[]>>()
+displayType<Immutable<Array<null>>>()
```
Comment (the-dr-lazy) [inline src/__tests__/types.dts.spec.ts:31]: Doesn't looks to be relevant.

File: src/__tests__/types.dts.spec.ts [hunk 7]
```diff
@@ -114,7 +114,7 @@ displayType<DeepImmutable<User>>()
 // @dts-jest:pass:snap
 displayType<DeepImmutable<undefined[]>>()
 // @dts-jest:pass:snap
-displayType<DeepImmutable<null[]>>()
+displayType<DeepImmutable<Array<null>>>()
```
Comment (the-dr-lazy) [inline src/__tests__/types.dts.spec.ts:117]: Doesn't looks to be relevant.

File: jest.config.js [hunk 8]
```diff
@@ -11,7 +11,7 @@ module.exports = {
     collectCoverageFrom: ['src/**/*.ts', '!src/**/*.spec.ts', '!**/__tests__/**'],
     globals: {
         'ts-jest': {
-            tsConfig: 'tsconfig.spec.json',
+            tsconfig: 'tsconfig.spec.json',
```
Comment (the-dr-lazy) [inline jest.config.js:14]: Doesn't looks to be relevant to me!

**Review by the-dr-lazy — CHANGES_REQUESTED** [review]
Thank you @Michael-1. I see lots of irrelevant changes. Are those by accidental?

File: package.json (hunk 4 above)
Comment (Michael-1) [reply package.json:74]: _@reduxjs/toolkit_ requires an update of _typescript,_ which in turn requires an update of _tslib_ and _ts-jest,_ which in turn requires an update of _jest._

The others aren’t strictly required for the switch to _@reduxjs/toolkit;_ they fix known security issues (as reported by `npm audit`).

File: src/__tests__/types.dts.spec.ts (hunk 6 above)
Comment (Michael-1) [reply src/__tests__/types.dts.spec.ts:31]: It’ required due to the Typescript update.

File: tsconfig.spec.json (hunk 5 above)
Comment (Michael-1) [reply tsconfig.spec.json:5]: It’s required due to the Typescript update.

File: CHANGELOG.md (hunk 1 above)
Comment (Michael-1) [reply CHANGELOG.md]: Okay, I’ve reverted it

File: jest.config.js (hunk 8 above)
Comment (Michael-1) [reply jest.config.js:14]: It’s required due to the `ts-jest` update.

File: tsconfig.spec.json (hunk 5 above)
Comment (Michael-1) [reply tsconfig.spec.json]: `esnext` is compatible with NodeJS 12 … but for now we can stay with it. I’ve reverted the change.

File: package.json (hunk 2 above)
Comment (Michael-1) [reply package.json]: Reverted.

File: package.json (hunk 3 above)
Comment (Michael-1) [reply package.json:55]: See below

**Comment by the-dr-lazy** [comment]
Due to changes in peerDependencies, this is a breaking change. I rather prefer to remove `configureStore` function and dependency to `@reduxt/toolkit` instead of maintaining it by a breaking change.

**Review by the-dr-lazy — APPROVED** [review]

**Pull request merged**